MosipNexus/deploy/nexus-server/install.sh

```
@@ -1,32 +1,120 @@
+        read -rs -p "Enter a PostgreSQL password for the 'mosip' user: " password
+        echo
+      done
+    else
+      # `|| true`: `head -c32` closing the pipe early SIGPIPEs `tr`, which
+      # pipefail would otherwise treat as a failure and abort the script.
+      password=$(LC_ALL=C tr -dc 'A-Za-z0-9' < /dev/urandom | head -c 32) || true
+      echo "Non-interactive shell and \$POSTGRES_PASSWORD unset — generated a random password (not printed)."
+    fi
+  fi
+
+  local pg_connection="postgresql+psycopg://mosip:$(urlencode "$password")@nexus-postgres:5432/mosipnexus"
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)